apps/designer/app/designer/shared/design-tokens-manager/design-tokens-manager.tsx

```diff
@@ -79,108 +104,230 @@ const TokenEditor = ({
-      onChangeComplete({ ...data, group, type } as DesignToken);
-      setIsOpen(false);
+      onChangeComplete({ ...token, ...data } as DesignToken);
+      onOpenChange(false);
+    }
+  };
+
+  const handleOpenChange = (isOpen: boolean) => {
+    if (isOpen === false) {
+      handleSubmit();
+      return;
     }
+    onOpenChange(isOpen);
   };
 
   return (
-    <Popover modal open={isOpen} onOpenChange={setIsOpen}>
+    <Popover modal open={isOpen} onOpenChange={handleOpenChange}>
       <PopoverTrigger
         asChild
         aria-label={isNew ? "Create Token" : "Edit Token"}
       >
-        <Button
-          ghost
-          onClick={(event) => {
-            event.preventDefault();
-            setIsOpen(true);
-          }}
-        >
-          <PlusIcon />
-        </Button>
+        {trigger ?? (
+          <Button
+            ghost
+            onClick={(event) => {
+              event.preventDefault();
+              onOpenChange(true);
+            }}
+          >
+            <PlusIcon />
+          </Button>
+        )}
       </PopoverTrigger>
       <PopoverPortal>
         <PopoverContent align="end" css={{ zIndex: "$zIndices$1" }}>
-          <form onChange={handleChange} onSubmit={handleSubmit}>
+          <form onChange={handleChange} onSubmit={handleSubmit} ref={formRef}>
             <Flex direction="column" gap="2" css={{ padding: "$spacing$7" }}>
               <Label htmlFor="name">Name</Label>
               <InputErrorsTooltip
                 errors={errors.name}
                 css={{ zIndex: "$zIndices$2" }}
               >
-                <TextField id="name" name="name" />
+                <TextField
+                  id="name"
+                  name="name"
+                  defaultValue={"name" in token ? token.name : ""}
+                />
               </InputErrorsTooltip>
               <Label htmlFor="value">Value</Label>
               <InputErrorsTooltip
                 errors={errors.value}
                 css={{ zIndex: "$zIndices$2" }}
               >
-                <TextField id="value" name="value" />
+                <TextField
+                  id="value"
+                  name="value"
+                  defaultValue={"value" in token ? token.value : ""}
+                />
               </InputErrorsTooltip>
               <Label htmlFor="description">Description</Label>
-              <TextArea id="description" name="description" />
+              <TextArea
+                id="description"
+                name="description"
+                defaultValue={"description" in token ? token.description : ""}
+              />
               {isNew && (
                 <Button type="submit" variant="blue">
                   Create
                 </Button>
               )}
             </Flex>
           </form>
-          <PopoverHeader title={isNew ? "New Token" : name} />
+          <PopoverHeader title={"name" in token ? token.name : "New Token"} />
```

Comment:
What is seed and why it doesn't have a name? Maybe make it undefined instead of optional? Then you can use nullish coalescing.

Reply:
yeah undefined could be an option, doesn't have a name because the token has group and type predefined but the rest is user-defined

Reply:
is it bad though the way it is, using in operator? What do you see as a problem here?

Reply:
separated seed from token so that there is no semantic confusion an no need for these checks